msgraphcore/middleware/retry_middleware.py

```diff
@@ -0,0 +1,184 @@
+import datetime
+import sys
+import time
+from email.utils import parsedate_to_datetime
+
+from msgraphcore.middleware.middleware import BaseMiddleware
+from msgraphcore.middleware.options.middleware_control import middleware_control
+
+
+class RetryMiddleware(BaseMiddleware):
+    """
+    TransportAdapter that allows us to specify the
+    retry policy for all requests
+    """
+
+    DEFAULT_TOTAL_RETRIES = 3
+    MAX_TOTAL_RETRIES = 10
+    DEFAULT_BACKOFF_FACTOR = 0.1
+    DEFAULT_RETRY_TIME_LIMIT = 180
+    MAXIMUM_BACKOFF = 120
+    _DEFAULT_RETRY_CODES = {429, 503, 504}
+
+    def __init__(self, retry_configs=None):
```

Comment:
This will break when a user initializes without passing an empty `retry_configs`. Is it possible to make the default value a dictionary with the default properties?